virttest/utils_misc.py

```diff
@@ -2230,6 +2231,7 @@ def program_is_alive(program_name, pid_files_dir=None):
     return pid_is_alive(pid)
 
 
+@utils_split_daemons.adopt_func_on_libvirtd
```

Comment:
it sounds like this method is also used by tp-qemu, so it is not proper to apply libvirtd related logic decorator on this method. One way to move forward is to copy similar function to utils lib  exclusively reserved for libvirt

Reply:
If so, I'm not quite understand what's the meaning for the comment in tp-libvirt PR 3813.

Reply:
@jinqi7 
Since signal_program is shared by both qemu and libvirt, so it is not proper to apply libvirt specific decorator here.
But I believe decorator adopt_func_on_libvirtd is still useful, so you can still use this PR to add adopt_func_on_libvirtd only.
Regarding  tp-libvirt PR 3813, there are two ways moving forward:
1)One is to port signal_program into your code in your tp-libvirt, then apply decorator on that
2)Two is to still use current signal_program in avocado-vt, but go back to use old way in previous tp-libvirt  implementation.

Reply:
@chunfuwen, For the first way,  do you mean keep duplicate functions in the script? Maybe it's not good.
For the second way, do you mean restore it back to the script below?  And also keep "adopt_func_on_libvirtd"  function in avocado-vt?
            if utils_split_daemons.is_modular_daemon():
                utils_misc.signal_program('libvirtd', 1,
                                      '/var/run')
           else:
               utils_misc.signal_program('virtqemud', 1,
                                      '/var/run')

Reply:
Yes, For second way, you can restore it back to the script as mentioned previously in tp-libvirt.

Reply:
Updated.  Shall we still keep the "adopt_func_on_libvirtd" without being used?@chunfuwen